lib/websocket/driver/hybi.rb

```diff
@@ -213,14 +213,26 @@ def send_frame(frame)
           buffer[2..9] = [length].pack(PACK_FORMATS[8]).bytes.to_a
         end
 
+        buffer = buffer.pack('C*')
+
         if frame.masked
-          buffer.concat(frame.masking_key.bytes.to_a)
-          buffer.concat(Mask.mask(frame.payload, frame.masking_key).bytes.to_a)
+          buffer << bin(frame.masking_key)
+          buffer << bin(Mask.mask(frame.payload, frame.masking_key))
         else
-          buffer.concat(frame.payload.bytes.to_a)
+          buffer << bin(frame.payload)
         end
 
-        @socket.write(buffer.pack('C*'))
+        @socket.write(buffer)
+      end
+
+
+      # For backwards compatibility with Rubies that lack {String#b} (Ruby 1.9).
+      def bin(str)
```

Comment:
This method copies the string data either by calling `b` or by doing the same operation ourselves. This is probably not necessary and could be optimised away.